plugins/module_utils/hcloud.py

```diff
@@ -91,6 +91,27 @@ def _build_client(self) -> None:
             application_version=version,
         )
 
+    def _client_get_by_name_or_id(self, resource: str, param: Union[str, int]):
+        """
+        Get a resource by name, and if not found by its ID.
+
+        :param resource: Name of the resource client that implements both `get_by_name` and `get_by_id` methods
+        :param param: Name or ID of the resource to query
+        """
+        resource_client = getattr(self.client, resource)
+
+        result = resource_client.get_by_name(param)
+        if result is not None:
+            return result
+
+        # If the param is not a valid ID, prevent an unnecessary call to the API.
+        try:
+            int(param)
+        except ValueError:
+            self.module.fail_json(msg=f"resource ({resource.rstrip('s')}) does not exist: {param}")
+
+        return resource_client.get_by_id(param)
```

Comment:
In the CLI & Terraform we first try the passed resource as ID and only if that fails we try to get it by name: https://github.com/hetznercloud/hcloud-go/blob/aea7836c36abaea253966b32bb1d114a8af548ab/hcloud/server.go#L227-L234

Doing it id->name has the benefit that it would be consistent with what is happening in hcloud-go.

Doing name->id has the benefit that its probably less of a breaking change for existing users, it also has the upside that you can use numerical names for your resources, which is not possible with the `.Get()` methods in `hcloud-go`.

Reply:
This change is not a new feature, but rather a consistency fix. We have other places where the behavior name>id is already implemented. So I would rather keep it consistent within our collection.